test/Sentry.Profiling.Tests/SamplingTransactionProfilerTests.cs

```diff
@@ -71,69 +73,83 @@ private SampleProfile CaptureAndValidate(ITransactionProfilerFactory factory)
         var clock = SentryStopwatch.StartNew();
         var hub = Substitute.For<IHub>();
         var transactionTracer = new TransactionTracer(hub, "test", "");
-        var sut = factory.Start(transactionTracer, CancellationToken.None);
+        var sut = factory.Start(transactionTracer, CancellationToken.None) as SamplingTransactionProfiler;
+        Assert.NotNull(sut);
         transactionTracer.TransactionProfiler = sut;
-        RunForMs(100);
+        RunForMs(RuntimeMs);
         sut.Finish();
         var elapsedNanoseconds = (ulong)((clock.CurrentDateTimeOffset - clock.StartDateTimeOffset).TotalMilliseconds * 1_000_000);
 
         var transaction = new Transaction(transactionTracer);
-        var collectTask = (sut as SamplingTransactionProfiler).CollectAsync(transaction);
+        var collectTask = sut.CollectAsync(transaction);
         collectTask.Wait();
         var profileInfo = collectTask.Result;
         Assert.NotNull(profileInfo);
         ValidateProfile(profileInfo.Profile, elapsedNanoseconds);
         return profileInfo.Profile;
     }
 
-    [Fact(Skip = "This test is flaky. TODO: rework")]
-    public Task Profiler_SingleProfile_Works()
+    [Fact]
+    public void Profiler_WithZeroStartupTimeout_CapturesAfterStartingAsynchronously()
     {
-        using var factory = SamplingTransactionProfilerFactory.Create(_testSentryOptions);
-        var profile = CaptureAndValidate(factory);
+        using var factory = new SamplingTransactionProfilerFactory(_testSentryOptions, TimeSpan.Zero);
+        var profiler = factory.Start(new TransactionTracer(Substitute.For<IHub>(), "test", ""), CancellationToken.None);
+        Assert.Null(profiler);
+        factory._sessionTask.Wait(60_000);
+        CaptureAndValidate(factory);
+    }
 
-        // We "Verify" part of a profile that seems to be stable.
-        var profileToVerify = new SampleProfile();
-        profileToVerify.Stacks.Add(profile.Stacks[0]);
-        for (var i = 0; i < profileToVerify.Stacks[0].Count; i++)
+    [Theory]
+    [InlineData(0)]
+    [InlineData(10)]
+    public void Profiler_SingleProfile_Works(int startTimeoutSeconds)
+    {
+        using var factory = new SamplingTransactionProfilerFactory(_testSentryOptions, TimeSpan.FromSeconds(startTimeoutSeconds));
+        // in the async startup case, we need to wait before collecting
+        if (startTimeoutSeconds == 0)
         {
-            var frame = profile.Frames[profileToVerify.Stacks[0][i]];
-            frame.Module = frame.Module.Replace("System.Private.CoreLib.il", "System.Private.CoreLib");
-            profileToVerify.Frames.Add(frame);
+            factory._sessionTask.Wait(60_000);
```

Comment:
Yikes... is this test flaky if we wait less than a minute for the _sessionTask?

Reply:
That's a new code required due to an introduced change: session is now started asynchronously. So in case it is, we need to wait before actually trying to collect profiles. 60 seconds is as good as any other time,

Reply:
And yet it can still fail to start up if the device is really overloaded... 
<img width="809" alt="image" src="https://github.com/getsentry/sentry-dotnet/assets/6349682/6bd84ffe-0a44-418e-aac0-538a2e24ee05">

I don't really know what to do with this other than allowing a startup failure when running in CI in which case the test would get skipped...

Reply:
Waits up to a minute for one test and still times out? This is def far from ideal. Skipping in CI is OK but is this going to be flaky locally? 
Hard to believe it can take more than 1 minute to startup

Reply:
I've made the change to skip tests on particular failing calls. [`44fec5a` (#2968)](https://github.com/getsentry/sentry-dotnet/pull/2968/commits/44fec5a3488cfba6dd75abfe0fe6d57f19534e63)